pilot/cmd/pilot-agent/request.go

```diff
@@ -15,25 +15,49 @@
 package main
 
 import (
+	"fmt"
 	"net/http"
 	"time"
 
 	"github.com/spf13/cobra"
 
+	"istio.io/istio/pkg/envoy"
+	"istio.io/pkg/env"
+
 	"istio.io/istio/pilot/pkg/request"
 )
 
+const defaultEnvyConfigPath = "/etc/istio/proxy/envoy-rev0.json"
+
+func init() {
+
+	requestCmd.PersistentFlags().StringVar(&meshConfigFile, "meshConfig", defaultMeshConfigFile,
+		"File name for Istio mesh configuration. If not specified, a default mesh will be used. This may be overridden by "+
+			"PROXY_CONFIG environment variable or proxy.istio.io/config annotation.")
+	proxyCmd.PersistentFlags().Int32Var(&envoyAdminPort, "envoyAdminPort", 0,
+		"Port where Envoy listens (on local host) for admin commands. If zero, get from mesh configuration or environment variable.")
+}
+
 // NB: extra standard output in addition to what's returned from envoy
 // must not be added in this command. Otherwise, it'd break istioctl proxy-config,
 // which interprets the output literally as json document.
 var (
+	envoyAdminPort int32
+
+	proxyAdminPortConfigEnv = env.RegisterIntVar(
+		"ISTIO_PROXY_ADMIN_PORT",
+		0,
+		"The proxy configuration. This will be set by the injection - gateways will use file mounts.",
```

Comment:
where? I cannot find this name anywhere from a grep of the code base

Reply:
This variable was added by me, I think it is possible to use environment variables to override the mesh configuration,I can remove it if it does n’t fit

Reply:
https://github.com/yjbdsky/istio/blob/master/pilot/cmd/pilot-agent/request.go#L85

Reply:
use this env in here:
https://github.com/yjbdsky/istio/blob/master/pilot/cmd/pilot-agent/request.go#L85